integration/tar_input_test.go

```diff
@@ -0,0 +1,304 @@
+		},
+		{
+			name: "centos 7 integration with low and high severity",
+			testArgs: args{
+				Version:       "dev",
+				SkipUpdate:    true,
+				IgnoreUnfixed: true,
+				Severity:      []string{"LOW", "HIGH"},
+				Format:        "json",
+				Input:         "testdata/fixtures/centos-7.tar.gz",
+			},
+			golden: "testdata/centos-7-low-high.json.golden",
+		},
+		{
+			name: "centos 6 integration",
+			testArgs: args{
+				Version:    "dev",
+				SkipUpdate: true,
+				Format:     "json",
+				Input:      "testdata/fixtures/centos-6.tar.gz",
+			},
+			golden: "testdata/centos-6.json.golden",
+		},
+		{
+			name: "ubi 7 integration",
+			testArgs: args{
+				Version:    "dev",
+				SkipUpdate: true,
+				Format:     "json",
+				Input:      "testdata/fixtures/ubi-7.tar.gz",
+			},
+			golden: "testdata/ubi-7.json.golden",
+		},
+		{
+			name: "distroless base integration",
+			testArgs: args{
+				Version:    "dev",
+				SkipUpdate: true,
+				Format:     "json",
+				Input:      "testdata/fixtures/distroless-base.tar.gz",
+			},
+			golden: "testdata/distroless-base.json.golden",
+		},
+		{
+			name: "distroless base integration with --ignore-unfixed option",
+			testArgs: args{
+				Version:       "dev",
+				SkipUpdate:    true,
+				IgnoreUnfixed: true,
+				Format:        "json",
+				Input:         "testdata/fixtures/distroless-base.tar.gz",
+			},
+			golden: "testdata/distroless-base-ignore-unfixed.json.golden",
+		},
+		{
+			name: "distroless python27 integration",
+			testArgs: args{
+				Version:    "dev",
+				SkipUpdate: true,
+				Format:     "json",
+				Input:      "testdata/fixtures/distroless-python27.tar.gz",
+			},
+			golden: "testdata/distroless-python27.json.golden",
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			// Copy DB file
+			cacheDir := gunzipDB()
+			defer os.RemoveAll(cacheDir)
+
+			// Setup CLI App
+			app := pkg.NewApp(c.testArgs.Version)
+			app.Writer = ioutil.Discard
+
+			osArgs := []string{"trivy", "--cache-dir", cacheDir, "--format", c.testArgs.Format}
+			if c.testArgs.SkipUpdate {
+				osArgs = append(osArgs, "--skip-update")
+			}
+			if c.testArgs.IgnoreUnfixed {
+				osArgs = append(osArgs, "--ignore-unfixed")
+			}
+			if len(c.testArgs.Severity) != 0 {
+				osArgs = append(osArgs,
+					[]string{"--severity", strings.Join(c.testArgs.Severity, ",")}...,
+				)
+			}
+			if len(c.testArgs.IgnoreIDs) != 0 {
+				trivyIgnore := ".trivyignore"
+				err := ioutil.WriteFile(trivyIgnore, []byte(strings.Join(c.testArgs.IgnoreIDs, "\n")), 0444)
+				assert.Nil(t, err, "failed to write .trivyignore")
+				defer os.Remove(trivyIgnore)
+			}
+			if c.testArgs.Input != "" {
+				osArgs = append(osArgs, []string{"--input", c.testArgs.Input}...)
+			}
+
+			// Setup the output file
+			var outputFile string
+			if *update {
+				outputFile = c.golden
+			} else {
+				output, _ := ioutil.TempFile("", "integration")
+				assert.Nil(t, output.Close())
```

Comment:
Why is the file closed here?

Reply:
I need just a temp file. The result of Trivy is exported to this file through pseudo CLI (`app.Run`) like this. `trivy --output /path/to/tempfile`
In short, `app.Run` writes the result to this file. We don't use this file descriptor directly. If the file is left open, an error occurs when Trivy (= `app.Run`) tries to open it.

Reply:
Gotcha. Thank you for the explanation!